src/ZZMatrix-linalg.jl

```diff
@@ -146,7 +143,7 @@ function _is_unimodular_given_det_mod_m(A::ZZMatrix, det_mod_m::Int, M::ZZRingEl
   # in CRT loop start with 22 bit primes; if we reach threshold (empirical), jump to much larger primes.
   p = 2^21; stride = 2^10; threshold = 5000000;
   A_mod_p = zero_matrix(Nemo.Native.GF(2), nrows(A), ncols(A))
-  while nbits(M) < Hbits
+  while nbits(M) <= Hbits
     @vprint(:UnimodVerif,2,".")
     # Next lines increment p (by random step up to stride) to a prime not dividing M
     if p > threshold && p < threshold+stride
```

Comment:
The `2^56` is two lines below this comment (I have no way to comment that line directly).
At least from the comment in that line it sounds like that magic number is dependent on you machine. And yes, Nemo should work on 32bit.
But I would leave it to @thofma to decide on what to do with this line as he originally brought this up.